src/analysis/complex/basic.lean

```diff
@@ -178,25 +178,37 @@ section real_deriv_of_complex
 open complex
 variables {e : ℂ → ℂ} {e' : ℂ} {z : ℝ}
 
-/-- If a complex function is differentiable at a real point, then the induced real function is also
-differentiable at this point, with a derivative equal to the real part of the complex derivative. -/
--- FIXME this causes a deterministic timeout with `-T50000`
-theorem has_deriv_at_real_of_complex (h : has_deriv_at e e' z) :
-  has_deriv_at (λx:ℝ, (e x).re) e'.re z :=
+/--
+A preliminary lemma for `has_deriv_at_real_of_complex`,
+which we only separate out to keep the maximum compile time per declaration low.
+-/
+lemma has_deriv_at_real_of_complex_aux (h : has_deriv_at e e' z) :
+  has_deriv_at (⇑continuous_linear_map.re ∘ λ {z : ℝ}, e (continuous_linear_map.of_real z))
```

Comment:
Can you use dot notation here please?

Reply:
Sorry, I'm not sure what you mean.

Reply:
Oh, you mean in the later lines. I managed to turn the first `continuous_linear_map.comp` into a `.comp`, but the second one, and the `restrict_scalars`, both cause typechecking errors when you convert them to dot notation. :-(

Reply:
Changing the associativity of the composition allows us to use one more dot, but still no luck with `restrict_scalars`.

Reply:
Ah, and adding the type annotation `(1 : ℂ →L[ℂ] ℂ)` allows us to write `.restrict_scalars`.

For clarity then I've added the type annotation on the other `1`.